jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -1508,11 +1561,57 @@ def run_extract1d(
     # Run the first integration to get the Tikhonov factor for the rest
     scidata = cube_model.data[0].astype("float64")
     scierr = cube_model.err[0].astype("float64")
-    scimask = np.bitwise_and(cube_model.dq[0], dqflags.pixel["DO_NOT_USE"]).astype(bool)
+    # Only mask pixels that are bad in *all* integrations
+    scimask = np.all(cube_model.dq & dqflags.pixel["DO_NOT_USE"] != 0, axis=0)
+
     refmask = bitfield_to_boolean_mask(
         cube_model.dq[0], ignore_flags=dqflags.pixel["REFERENCE_PIXEL"], flip_bits=True
     )
 
+    # Pixels that are bad in all integrations
+
+    allbad = np.sum(np.isfinite(cube_model.err) &
+                    np.isfinite(cube_model.data), axis=0) == 0
+
+    # Make a copy of the cube model where bad pixels in any individual
+    # integration are replaced with a running median in time.
+
+    data_nanreplaced = cube_model.data*1.
+    data_infilled = cube_model.data*1.
```

Comment:
Are these copies necessary?  I don't think the cubemodel data is 


If they are necessary, is there any reason you can't just use `array.copy()`?  Or use `np.empty_like` to initialize an empty one?

Reply:
But I don't actually see where `data_nanreplaced` gets modified.  It looks like you do the actual replacement into `data_infilled`